kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -189,6 +195,333 @@ object StreamCoreExtensions:
+                                                    else if newRemainingPar <= 0 then
+                                                        nextChunkEffect.map: nextChunk =>
+                                                            prevChunkFiber.get.map: prevChunk =>
+                                                                prevEmitFiber.get.andThen:
+                                                                    Loop.continue(
+                                                                        Fiber.success(prevChunk ++ nextChunk),
+                                                                        newRemainingChunk,
+                                                                        parallel
+                                                                    )
+                                                    else if newRemainingChunk.size <= 0 then
+                                                        Async.run {
+                                                            nextChunkEffect.map: nextChunk =>
+                                                                prevChunkFiber.get.map: prevChunk =>
+                                                                    prevEmitFiber.get.andThen:
+                                                                        val chunk = prevChunk ++ nextChunk
+                                                                        channel.put(Present(chunk)).andThen:
+                                                                            parAdjustmentRef.updateAndGet(_ + chunk.size).unit
+                                                        }.map: nextFiber =>
+                                                            Loop.done(((nextFiber, newRemainingPar), cont(())))
+                                                    else
+                                                        bug("Illegal state: there is remaining parallel and remaining chunk in mapPar")
+                                                    end if
+                                ,
+                                done = {
+                                    case ((lastFiber, _), _) =>
+                                        lastFiber.get.andThen:
+                                            channel.put(Absent)
+                                }
+                            )
+
+                            Abort.fold[E | Closed](
+                                _ => Abort.run(channel.put(Absent)).unit,
+                                {
+                                    case _: Closed       => bug("buffer closed unexpectedly")
+                                    case e: E @unchecked => Abort.run(channel.put(Absent)).andThen(Abort.fail(e))
+                                },
+                                e => Abort.run(channel.put(Absent)).andThen(Abort.panic(e))
+                            )(handledStream)
```

Comment:
readability improvement
```suggestion
                            Abort.fold[E | Closed](
                                onSuccess = _ => Abort.run(channel.put(Absent)).unit,
                                onFail = {
                                    case _: Closed       => bug("buffer closed unexpectedly")
                                    case e: E @unchecked => Abort.run(channel.put(Absent)).andThen(Abort.fail(e))
                                },
                                onPanic = e => Abort.run(channel.put(Absent)).andThen(Abort.panic(e))
                            )(handledStream)
```